assets/pages/regular-contributions/helpers/ajax.js

```diff
@@ -89,38 +89,25 @@ const paymentMethodToPaymentFieldMap = {
 };
 
 const getPaymentFields =
-  (
-    token?: string,
-    accountNumber?: string,
-    sortCode?: string,
-    accountHolderName?: string,
-    paymentFieldName: string,
-  ): ?(PayPalDetails | StripeDetails | DirectDebitDetails
-    ) => {
+  (token: Token): ?(PayPalDetails | StripeDetails | DirectDebitDetails) => {
```

Comment:
Can we do without this method (and possibly the types `PaypalDetails`, `StripeDetails` and `DirectDebitDetails`), since they seem to capture much the same information as `Token`?

Reply:
actually I've realised there there is some mapping between field names (token => baid for PayPal, token => stripeToken for stripe), maybe because of backend requirements. I agree it seems pointless and could probably be simplified but there's enough in this PR I think